misc,tests,github: Use pre-commit CI over pre-commit actions; cleanup pre-commit and workflow files

BobbyRBruce opened the pull request:
This is some stuff I've been playing with and finally have working.

The main contribution here is moving to pre-commit CI: https://pre-commit.ci. The main advantage of pre-commit CI is it's a GitHub app and will automatically push a patch to broken PRs.

Since i was already playing with pre-commit I updated our YAML formatter to something superior and from that I cleanup up a lot of our Github Workflow actions: most just improving docstring and removing pointless steps and minor inefficiencies.

BobbyRBruce committed:
python,tests,misc: Replace Action precommit with CI

This change removes the GitHub Action Job which ran pre-commit with the
pre-commit CI GitHub App (see https://pre-commit.ci for more information
on this).

The advantage of this approach is the GitHub App can push changes which
automatically fix pre-commit errors to the PR.

BobbyRBruce committed:
misc,base: Add errors for pre-commit CI to autofix

Change-Id: I55f339d66ebf6594baca39a2b268c00ed7930b1f

pre-commit-ci[bot] committed:
misc: Automatic fixes from pre-commit hooks

This is an automatic commit generated by the pre-commit CI
process to automatically fix detected errors

For more information, see https://pre-commit.ci.

BobbyRBruce committed:
python,tests,misc: Update yaml pre-commit and format files

Replaces the yaml formatter
https://github.com/jumanjihouse/pre-commit-hook-yamlfmt with
https://github.com/macisamuele/language-formatters-pre-commit-hooks

The older YAML pre-commit appears to be a dead project and never worked
in a satisfying way. This upgrade gives a superior experience.

Change-Id: If7a21475dd53cf0afeb96676d16542483877fce0

BobbyRBruce committed:
util-docker: Add 'jq' to 24.04 all-dep Docker image

This tool is used in the testing script. It should be included in the
Docker image, not installed each time.

Change-Id: I7565dec3f72caeeb8b752ee0943e00e89121b654

BobbyRBruce committed:
misc,tests,python: Refactor/cleanup the GitHub Workflows

General cleanup and refactor. Tests are functionally the same but
redundancy has been removed, comments improved, and things formatted
better.

Generally this is a cleanup of the workflow files since introducing the
new YAML formatter. There is now less indirection and the comments have
been improved and made clearer where possible.

Change-Id: I04d3788805b808de5a24e4574cb6608d0d13bad6

powerjg reviewed (commented):
This looks good to me

BobbyRBruce committed:
misc: Fix chmod xargs in Workflows

Change-Id: I5c2834603df593a8c4b6516acfeaa0c7ae72e8e3

BobbyRBruce committed:
misc: Fix upload

Change-Id: I4622441785db2c2411c134a2ab805a86bfd4d917

BobbyRBruce committed:
misc,tests,gpu-compute: Move some GPU tests to CI

BobbyRBruce committed:
gpu-compute,tests: Move GPU tests to testlib

A new host tag `gcn_gpu` has been added. This allows for selection of
those GPU tests which depend upon the gcn-gpu docker image to run.

In addition to this, the square GPU tests has been moved to the CI
tests. This ensures some GPU code is compiled and run on every PR.

Change-Id: Ic1a3e3a5c14566c5a53aeb74ab6d0725a1310fff

BobbyRBruce committed:
misc,tests,gpu-compute:"Revert Move some GPU tests to CI"

This reverts commit e659646bcbe6163d85fd42ed43173a22e993c7d3.

BobbyRBruce commented:
I'm going to put this on hold until #1270 and #1292 are merged. This touches too much and I don't want to disrupt.

BobbyRBruce committed:
misc,tests: Fix bad merge in ci-tests.yaml

BobbyRBruce committed:
misc,tests: Add missing '}' to GH Workflow job name

BobbyRBruce committed:
misc,tests: Fix wrong job names in ci-tests.yaml

BobbyRBruce committed:
misc,tests: Fix merge error in daily-tests.yaml & formatting

BobbyRBruce committed:
tests,misc: Added gpu-tests and get-date draft guard

BobbyRBruce committed:
tests,misc: Typo and formatting fixes to ci-tests.yaml

BobbyRBruce committed:
misc,tests: Minor changes (typos, formatting) to workflows

BobbyRBruce committed:
util-docker: Remove Docker-comppse.yaml

This was accidently re-added. Removing.

BobbyRBruce committed:
util-docker: Remove 'js' install form 24.04_all-deps

This is no longer needed.

BobbyRBruce committed:
misc,tests: Remove retention period decl

BobbyRBruce committed:
misc,tests: v4.0.0.0 to v4 for downloader action in weeklies

BobbyRBruce committed:
tests,misc: Don't run 'get-date' job if PR is draft

BobbyRBruce committed:
misc,tests: 'Conformation' -> 'Confirmation' spelling fix

BobbyRBruce committed:
misc,tests: Get cached VEGA build in CI Tests

BobbyRBruce committed:
misc,tests: Make 'get-date' run on GH Hosted Runner

BobbyRBruce committed:
misc,tests: Rename jobs/steps

BobbyRBruce committed:
misc,tests: Fix matrix job 'needs'

erin-le commented:
@BobbyRBruce: This PR is being closed for inactivity.
